tests/validate_example_current_schema.py

```diff
@@ -1,38 +1,44 @@
-    # We need to serve the current "camtrap-dp-profile.json" over an URL so it can be referenced in the descriptor
-    httpd, address = ServeDirectoryWithHTTP(CURRENT_PACKAGE_PROFILE_PATH.parent)
-    fixed_address = address.replace("1.0.0.127.in-addr.arpa", "localhost")
-
-    with open(EXAMPLE_DESCRIPTOR_PATH) as json_file:
-        print(f"Validating {EXAMPLE_DESCRIPTOR_PATH} against current schemas")
+    # Profile and schemas are referenced by URL in the descriptor (datapackage.json)
+    # Here we setup a local server to host the current profile and schemas
+    httpd, address = ServeDirectoryWithHTTP(PROFILE_PATH.parent)
+    server_address = address.replace("1.0.0.127.in-addr.arpa", "localhost")
+
+    # The profile file has a requirement that schema URLs should contain a version number
+    # Here we create a "relaxed" profile file where that requirement is changed
+    shutil.copyfile(PROFILE_PATH, RELAXED_PROFILE_PATH)
+    data = RELAXED_PROFILE_PATH.read_text()
+    data = re.sub('"pattern": ".*"', '"pattern": "localhost"', data) # Don't require version number, require "localhost"
+    RELAXED_PROFILE_PATH.write_text(data)
+
+    with open(EXAMPLE_PATH) as json_file:
+        print(f"Validating {EXAMPLE_PATH} against current schemas")
         descriptor_data = json.load(json_file)
 
-        # In this case, we update the descriptor so it points to the current version
-        # 1. Replace profile URL with current one
-        descriptor_data['profile'] = f"{fixed_address}/camtrap-dp-profile.json"
-        # 2. Replace resources schemas with current one
-        for resource_to_patch in RESOURCES_TO_PATCH:
-            resource_index = next((i for i, item in enumerate(descriptor_data['resources']) if item["name"] == resource_to_patch['resource_name']), None)
-            with open(resource_to_patch['current_schema_path']) as schema_file:
-                descriptor_data['resources'][resource_index]['schema'] = json.load(schema_file)
+        # 1. Replace profile URL with "relaxed" local one
+        descriptor_data['profile'] = f"{server_address}/{RELAXED_PROFILE_PATH.name}"
+
+        # 2. Replace schema URLs of the resources with local ones
+        for table_schema in TABLE_SCHEMA_PATHS:
+            resource_index = next((i for i, item in enumerate(descriptor_data['resources']) if item["name"] == table_schema.name.split("-")[0]), None)
+            descriptor_data['resources'][resource_index]['schema'] = f"{server_address}/{table_schema.name}"
 
         # Frictionless loads resource relative to the current directory, (and not to the descriptor)
         # Issue reported to frictionless: https://github.com/frictionlessdata/frictionless-py/issues/956
-        os.chdir(EXAMPLE_DESCRIPTOR_PATH.parent)
+        os.chdir(EXAMPLE_PATH.parent)
 
         validate_package_print_and_exit(descriptor_data)
+
+        # Remove "relaxed" profile
+        # TODO: doesn't get deleted
+        RELAXED_PROFILE_PATH.unlink()
```

Comment:
@niconoe could use your Python experience here :smile:

This line does remove the file, but not when it's in or after the `with:` statement. How would you tackle this?